packager/certfetcher/certfetcher.go

```diff
@@ -54,8 +55,10 @@ func (u *AcmeUser) GetPrivateKey() crypto.PrivateKey {
 }
 
 // Initializes the cert fetcher with information it needs to fetch new certificates in the future.
-func NewFetcher(email string, privateKey crypto.PrivateKey, acmeDiscoURL string,
-	domains []string, acmeChallengePort int, shouldRegister bool) (*CertFetcher, error) {
+func New(email string, certSignRequest *x509.CertificateRequest, privateKey crypto.PrivateKey,
+	acmeDiscoURL string, httpChallengePort int, httpChallengeWebRoot string,
+	tlsChallengePort int, dnsProvider string, shouldRegister bool) (*CertFetcher, error) {
```

Comment:
optional again. Same thought here about breaking this up. Callsite could have some structure like:

fetcher := CertFetcher()
fetcher.setUser(email, privateKey)
fetcher.bindToPort(port)

Reply:
Added TODO.